### PR TITLE
Add HasExact to topology endpoint

### PR DESCRIPTION
**Pull request by freddygv**
This flag is being added so that the UI can know whether an exact intention exists between two services. 

If an exact intention exists users will be sent to an "edit" page to avoid overwriting an existing intention.

Now the Intention object attached to each upstream/downstream in the response will look like this:

```
"Intention": {
  "Allowed": false,
  "HasPermissions": true,
  "HasExact: true
}
```

### CODE REVIEW

File: agent/structs/intention.go [hunk 1]
```diff
@@ -26,6 +26,8 @@ const (
 	// fix up all the places where this was used with the proper namespace
 	// value.
 	IntentionDefaultNamespace = "default"
+
+	MaxIntentionPrecedence = 9
```
Comment (rboyer) [inline agent/structs/intention.go:30]: Did you mean to use this instead of the wildcard check?

Comment (freddygv) [reply agent/structs/intention.go]: Originally, yeah, but then remembered config intentions don't have a precedence.

Comment (freddygv) [reply agent/structs/intention.go]: Removed this.

**Review by rboyer — APPROVED** [review]
LGTM

**Pull request merged**